async-aggregators/src/control_panel.py

```diff
@@ -0,0 +1,225 @@
+        self.setWidgetResizable(True)
+        content = QWidget(self)
+        self.setWidget(content)
+        lay = QVBoxLayout(content)
+        self.label = QLabel(content)
+        self.label.setWordWrap(True)
+        self.label.setAlignment(Qt.AlignmentFlag.AlignCenter)
+        lay.addWidget(self.label)
+
+    def setText(self, text: str) -> None:
+        """
+        Set the text of the label
+
+        :param text: The text to set
+        :return:
+        """
+        self.label.setText(text)
+
+
+class ControlPanel(QWidget):
+    """Control Panel"""
+
+    # Define a new signal at the top of the class
+    sliderValueChanged = pyqtSignal(str, int)
+    comboBoxesSwapped = pyqtSignal(str, str)
+    ascii = pyqtSignal()
+
+    def __init__(self, title: str, widget_info: dict):
+        super().__init__()
+        layout = QVBoxLayout(self)
+        self.title = title
+        self.combo_boxes = []
+        self.description = widget_info.get("description", None)
+
+        title_box = self.create_panel_title(title, self.description)
+        layout.addWidget(title_box)
+
+        for info in widget_info.get("sliders", []):
+            label, tickMarkLabels, slider_range, orientation, disable_mouse_drag = info
+            layout.addWidget(QLabel(label))
+            if tickMarkLabels:
+                slider = QCustomSlider(
+                    0,
+                    20,
+                    1,
+                    orientation,
+                    labels=tickMarkLabels,
+                    suppress_mouse_move=disable_mouse_drag,
+                )
+            else:
+                slider = QCustomSlider(0, 20, 1, orientation)
+            slider.sl.valueChanged.connect(
+                lambda value, lbl=label: self.forward_signal(lbl, value)
+            )
+            slider_frame = self.style_slider(
+                slider, slider_range, orientation == Qt.Orientation.Horizontal
+            )
+            slider_frame.setMaximumHeight(60)
+            layout.addWidget(slider_frame)
+        # Adding dropdowns
+        for info in widget_info.get("dropdowns", []):
+            combo_box1 = QComboBox()
+            combo_box2 = QComboBox()
+
+            combo_box1.addItems(info)
+            combo_box2.addItems(info)
+
+            layout.addWidget(combo_box1)
+            layout.addWidget(combo_box2)
+
+            # Append the QComboBoxes to the list
+            self.combo_boxes.append((combo_box1, combo_box2))
+        # Adding combo box buttons
+        for button_text in widget_info.get("combo_box_buttons", []):
+            button = QPushButton()
+            button.setText(button_text)
+            # Use default arguments in the lambda to capture the current values of combo_box1 and combo_box2
+            # Assuming one button per pair of dropdowns.
+            cb1, cb2 = self.combo_boxes.pop(
+                0
+            )  # Pop the first pair of QComboBoxes from the list
```

Comment:
Don't let your formatter be your enemy. (Also this comment could really just be deleted since it doesn't really add anything.)
```suggestion
            # Pop the first pair of QComboBoxes from the list
            cb1, cb2 = self.combo_boxes.pop(0)
```